### PR TITLE
Herbivores and carnivores task complete

### PR DESCRIPTION
**Pull request by thorium-zero**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,50 @@
...
+            return
+        self._health -= value
+        if self._health < 1:
+            Animal.alive.remove(self)
+
+
+class Carnivore(Animal):
+    def bite(self, prey: Herbivore) -> None:
+        if not isinstance(prey, Herbivore) or prey.hidden:
+            return
```
Comment (YuriiKindrat) [inline app/main.py:49]: no need to use blank return here, u can use backward condition for this

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (thorium-zero) [reply app/main.py:49]: what's wrong with empty returns? oO

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]
Remove the empty line and rewrite without protected arguments, please 🌺 You can read an article about it: [The Meaning of Underscores in Python](The Meaning of Underscores in Python) 🐥

File: app/main.py [hunk 2]
```diff
@@ -1 +1,53 @@
...
+from __future__ import annotations
+
+
+class Animal:
+    alive: list[Animal] = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 is_hidden: bool = False) -> None:
+        self._name: str = name
```
[the remark this answers is not in the document]
Comment (thorium-zero) [reply app/main.py:10]: I can't do that. Not in this lifetime at least :D

**Review by LiudmylaKulinchenko — APPROVED** [review]
Good job 🍬🔥🚀

**Pull request closed without merging**